acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -224,13 +227,25 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
     /// Resolves a foreign call's [result][acir::brillig_vm::ForeignCallResult] using a result calculated outside of the ACVM.
     ///
     /// The ACVM can then be restarted to solve the remaining Brillig VM process as well as the remaining ACIR opcodes.
-    pub fn resolve_pending_foreign_call(&mut self, foreign_call_result: ForeignCallResult) {
+    pub fn resolve_pending_foreign_call(&mut self, foreign_call_result: ACVMForeignCallResult) {
         if !matches!(self.status, ACVMStatus::RequiresForeignCall(_)) {
             panic!("ACVM is not expecting a foreign call response as no call was made");
         }
 
         let brillig_solver = self.brillig_solver.as_mut().expect("No active Brillig solver");
-        brillig_solver.resolve_pending_foreign_call(foreign_call_result);
+        match foreign_call_result {
+            ACVMForeignCallResult::BrilligOutput(foreign_call_result) => {
+                brillig_solver.resolve_pending_foreign_call(foreign_call_result);
+            }
+            ACVMForeignCallResult::ResolvedAssertMessage(assert_message) => {
+                if assert_message.is_empty() {
```

Comment:
In order to reset the current assertion message I went with inserting a resolve_assert_message call before every constrain and in the case one isn't specified we simply have an empty string. However, I could change this to be a separate boolean flag or something but I didn't think it was very important people got to print empty strings.